### PR TITLE
[TRAFFIC-9318] Add `confluent network peering create` command

### PR DESCRIPTION
**Pull request by syoung-cflt**
What
----

* Added the `addNetworkFlag` function with autocomplete support (this will be used in most of the create commands)
* Updated examples in `confluent network create` and the relevant test golden files
* Added `confluent network peering create` to create a peering connection.
* Updated linter exclude with the newly added flags

References
----------

[CLI Commands Design](https://confluentinc.atlassian.net/wiki/spaces/PM/pages/2987394300/1-pager+--+Network+CLI)
[Implementation 1-Pager](https://confluentinc.atlassian.net/wiki/spaces/~712020463029eabbac401bbe1b033ce0d3a00e/pages/3136749631/Network+CLI)

Test & Review
-------------

Added integration tests.

Performed manual tests

Open Questions / Follow-ups
---------------------------

### CODE REVIEW

File: internal/network/command_peering_create.go [hunk 1]
```diff
@@ -0,0 +1,212 @@
...
+	pcmd.AddCloudFlag(cmd)
+	cmd.Flags().String("aws-account", "", "AWS account ID associated with the VPC that you are peering with Confluent Cloud network.")
+	cmd.Flags().String("gcp-project", "", "Google Cloud project ID associated with the VPC that you are peering with Confluent Cloud network.")
+	cmd.Flags().String("azure-tenant", "", "Azure Tenant ID in which your Azure Subscription exists.")
+	cmd.Flags().String("aws-vpc", "", "AWS VPC ID that you are peering with Confluent Cloud network.")
+	cmd.Flags().String("gcp-vpc-network", "", "Name of the Google Cloud VPC that you are peering with Confluent Cloud network.")
+	cmd.Flags().String("azure-vnet", "", "Azure Resource ID of the VNet that you are peering with Confluent Cloud.")
+	cmd.Flags().StringSlice("aws-routes", nil, "A comma-separated list of CIDR blocks of the AWS VPC that you are peering with Confluent Cloud network.")
+	cmd.Flags().Bool("gcp-import-custom-routes", false, "Enable customer route import for Google Cloud VPC Peering.")
+	cmd.Flags().String("customer-region", "", "Cloud region ID of the AWS VPC or Azure VNet that you are peering with Confluent Cloud network.")
```
Comment (syoung-cflt) [inline internal/network/command_peering_create.go:46]: @kabraham13 Could you please review these flags and examples? 🙏

File: cmd/lint/main.go [hunk 2]
```diff
@@ -121,6 +122,8 @@ var flagRules = []linter.FlagRule{
 			"destination-api-key",
 			"destination-api-secret",
 			"destination-bootstrap-server",
+			"gcp-import-custom-routes",
```
Comment (brianstrauch) [inline cmd/lint/main.go:125]: Any way we can shorten this name?

File: internal/network/command_peering_create.go [hunk 3]
```diff
@@ -0,0 +1,212 @@
...
+	addNetworkFlag(cmd, c.AuthenticatedCLICommand)
+	pcmd.AddCloudFlag(cmd)
+	cmd.Flags().String("aws-account", "", "AWS account ID associated with the VPC that you are peering with Confluent Cloud network.")
+	cmd.Flags().String("gcp-project", "", "Google Cloud project ID associated with the VPC that you are peering with Confluent Cloud network.")
+	cmd.Flags().String("azure-tenant", "", "Azure Tenant ID in which your Azure Subscription exists.")
+	cmd.Flags().String("aws-vpc", "", "AWS VPC ID that you are peering with Confluent Cloud network.")
+	cmd.Flags().String("gcp-vpc-network", "", "Name of the Google Cloud VPC that you are peering with Confluent Cloud network.")
+	cmd.Flags().String("azure-vnet", "", "Azure Resource ID of the VNet that you are peering with Confluent Cloud.")
+	cmd.Flags().StringSlice("aws-routes", nil, "A comma-separated list of CIDR blocks of the AWS VPC that you are peering with Confluent Cloud network.")
+	cmd.Flags().Bool("gcp-import-custom-routes", false, "Enable customer route import for Google Cloud VPC Peering.")
```
Comment (brianstrauch) [inline internal/network/command_peering_create.go:45]: Would it make sense to put the AWS flags next to each other?

File: internal/network/command_peering_create.go [hunk 4]
```diff
@@ -0,0 +1,212 @@
...
+	if err != nil {
+		return nil, err
+	}
+
+	vpcNetwork, err := cmd.Flags().GetString("gcp-vpc-network")
+	if err != nil {
+		return nil, err
+	}
+
+	importCustomRoutes, err := cmd.Flags().GetBool("gcp-import-custom-routes")
```
Comment (brianstrauch) [inline internal/network/command_peering_create.go:170]:
```suggestion
	gcpImportCustomRoutes, err := cmd.Flags().GetBool("gcp-import-custom-routes")
```

File: internal/network/command_peering_create.go (hunk 1 above)
Comment (syoung-cflt) [reply internal/network/command_peering_create.go]: Synced with @kabraham13 and @brianstrauch. We are condensing the flags as follows 
`cloud-account`: `aws-account`, `gcp-project`, and `azure-tenant`
`virtual-network`: `aws-vpc`, `gcp-vpc-network`, and `azure-vnet`

File: cmd/lint/main.go (hunk 2 above)
Comment (syoung-cflt) [reply cmd/lint/main.go]: checked with @kabraham13 and shortened it to `gcp-routes`

File: internal/network/command_peering_create.go (hunk 4 above)
Comment (syoung-cflt) [reply internal/network/command_peering_create.go]: Updated!

File: internal/network/command_create.go [hunk 5]
```diff
@@ -21,19 +21,19 @@ func (c *command) newCreateCommand() *cobra.Command {
 		Example: examples.BuildExampleString(
 			examples.Example{
 				Text: `Create a Confluent network in AWS with connection type "transitgateway" by specifying zones and CIDR.`,
-				Code: "confluent network create aws-tgw --cloud aws --region us-west-2 --connection-types transitgateway --zones usw2-az1,usw2-az2,usw2-az4 --cidr 10.1.0.0/16",
+				Code: "confluent network create aws-tgw-network --cloud aws --region us-west-2 --connection-types transitgateway --zones usw2-az1,usw2-az2,usw2-az4 --cidr 10.1.0.0/16",
```
Comment (brianstrauch) [inline internal/network/command_create.go:24]: Add an example where `--connection-types` has multiple values?

File: internal/network/command_peering_create.go [hunk 6]
```diff
@@ -0,0 +1,211 @@
...
+func (c *peeringCommand) newCreateCommand() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "create <name>",
+		Short: "Create a peering.",
+		Args:  cobra.ExactArgs(1),
+		RunE:  c.create,
+		Example: examples.BuildExampleString(
+			examples.Example{
+				Text: "Create an AWS VPC peering.",
+				Code: "confluent network peering create aws-peering --network n-abcde1 --cloud aws --cloud-account 012345678901 --virtual-network vpc-abcdef0123456789a --aws-routes 172.31.0.0/16,10.108.16.0/21",
```
Comment (brianstrauch) [inline internal/network/command_peering_create.go:23]: Are you sure that vpc-xyz normally has 17 trailing characters? Seems like a strange length...
```suggestion
				Code: "confluent network peering create aws-peering --network n-123456 --cloud aws --cloud-account 012345678901 --virtual-network vpc-0123456789abcdef0 --aws-routes 172.31.0.0/16,10.108.16.0/21",
```

Comment (syoung-cflt) [reply internal/network/command_peering_create.go:23]: yup... 

Found [this article](https://aws.amazon.com/about-aws/whats-new/2018/02/longer-format-resource-ids-are-now-available-in-amazon-ec2/) from 2018 

> The current format is a resource identifier followed by an 8-character string. The new format will be the same resource identifier followed by a 17-character string. For example, today, your VPC has a resource identifier like “vpc-1234abc0”. After July 2018, a new VPC will be assigned an identifier like “vpc-1234567890abcdef0”.

Comment (brianstrauch) [reply internal/network/command_peering_create.go:23]: Nice, updated the suggestion to follow their format.

Comment (syoung-cflt) [reply internal/network/command_peering_create.go]: Updated!

File: internal/network/command_create.go (hunk 5 above)
Comment (syoung-cflt) [reply internal/network/command_create.go:24]: Added!

**Review by brianstrauch — APPROVED** [review]

**Pull request merged**